userspace/libscap/scap_open.h

```diff
@@ -78,16 +77,6 @@ extern "C"
 		bool ppm_sc[PPM_SC_MAX];
 	} interesting_ppm_sc_set;
 
-	/*!
-	 * \brief Argument for scap_open
-	 * Set any tracepoint idx to true to enable its tracing at driver level,
-	 * otherwise a tp is not attached (so called "uninteresting tracepoint").
-	 */
-	typedef struct
-	{
-		bool tp[TP_VAL_MAX];
-	} interesting_tp_set;
```

Comment:
I have a side question: in cases like the old probe where we have a dedicated tracepoint like `sched_proc_fork`  used only in this engine, how can we address this issue?

https://github.com/falcosecurity/libs/blob/0c405c739dfa7e7cef4f07d12ac47468f17f253d/driver/bpf/probe.c#L204

I see 2 main ways:

1. Every engine has its own table of tracepoints and according to ppm_sc is able to understand which tracepoints to enable. In this case, the choice to put the conversion logic inside every engine (like in this PR) seems reasonable but we need a dedicated table for every engine while in this PR we have just one shared table.

2. We have a single tracepoint enum shared between all drivers, and in this case, we will add just a new code like `SCHED_PROG_FORK_OLD_PROBE` and simply all the other engines will ignore this code because they don't know it. In this case, probably we don't need a specific conversion logic for every engine like in this PR and we can keep this `interesting_tp_set` struct, converting ppm_sc to tp outside the engines.

I don't want to solve this corner case in this PR but according to how we want to address the problem I think we need to take a different road, WDYT? Maybe there are some other alternatives but right now I have only these 2 on top of my mind :/

Reply:
I'd go with 2; if the differences keep growing, we can go back at 1 in the future .

Reply:
ok, let's now keep it as it is and maybe I will address this corner case in another PR proposing a slightly different approach :)